lib/active_model_serializers/adapter/attributes.rb

```diff
@@ -1,13 +1,32 @@
 module ActiveModelSerializers
   module Adapter
     class Attributes < Base
+      def initialize(*)
+        super
+        instance_options[:fieldset] ||= ActiveModel::Serializer::Fieldset.new(fields_to_fieldset(instance_options.delete(:fields)))
+      end
+
       def serializable_hash(options = nil)
         options = serialization_options(options)
         options[:fields] ||= instance_options[:fields]
         serialized_hash = serializer.serializable_hash(instance_options, options, self)
 
         self.class.transform_key_casing!(serialized_hash, instance_options)
       end
+
+      def fields_to_fieldset(fields)
+        return fields if fields.nil?
+        resource_fields = []
+        relationship_fields = {}
+        fields.each do |field|
+          case field
+          when Symbol, String then resource_fields << field
+          when Hash then relationship_fields.merge!(field)
+          else fail ArgumentError, "Unknown conversion of fields to fieldset: '#{field.inspect}' in '#{fields.inspect}'"
+          end
+        end
+        relationship_fields.merge(serializer.json_key.to_sym => resource_fields)
```

Comment:
Could please change it either to `merge!` or 
```ruby
relationship_fields[serializer.json_key.to_sym] = resource_fields
relationship_fields
```

Reply:
good point. I wonder why I didn't mutate a local var we own and mutate elsewhere

@wasifhossain you're welcome to take ownership of this PR if you'd like

Reply:
@bf4 thanks for the offer. I will rebase it as @cintamani suggested earlier as well as make this small fix (prefer using `merge!`) before merging this PR. do you have any suggestion, pls?

Reply:
Only real suggestion is to bear in mind once merged we need to support it.

I didn't merge it in part due to lack of confidence it was the right change. 

I don't like to add features without someone actually trying it out.